app/services/samples/clone_service.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Samples
+  # Service used to clone samples
+  class CloneService < BaseProjectService
+    CloneError = Class.new(StandardError)
+
+    def execute(new_project_id, sample_ids)
+      validate(new_project_id, sample_ids)
+
+      authorize! @project, to: :clone_sample?
+
+      @new_project = Project.find_by(id: new_project_id)
+      authorize! @new_project, to: :clone_sample_into_project?
+
+      clone_samples(@new_project, sample_ids)
+    rescue Samples::CloneService::CloneError => e
+      @project.errors.add(:base, e.message)
+      {}
+    end
+
+    private
+
+    def validate(new_project_id, sample_ids)
+      raise CloneError, I18n.t('services.samples.clone.empty_new_project_id') if new_project_id.blank?
+
+      raise CloneError, I18n.t('services.samples.clone.empty_sample_ids') if sample_ids.blank?
+
+      return unless @project.id == new_project_id.to_i
+
+      raise CloneError, I18n.t('services.samples.clone.same_project')
+    end
+
+    def clone_samples(new_project, sample_ids)
+      cloned_sample_ids = {}
+      sample_ids.each do |sample_id|
+        sample = Sample.find_by(id: sample_id, project_id: @project.id)
+        clone = Sample.new(name: sample.name, description: sample.description, project_id: new_project.id)
+        Samples::Metadata::UpdateService.new(new_project, clone, @current_user,
```

Comment:
Any errors thrown by the sample metadata update service would just be ignored. We need to catch if there is are any errors and attach them to the `project.errors`

Reply:
I'm assuming we want to check if there are any not_updated changes sent back, like we did for the metadata file import service? Please see 92a72016105c65a3c2aa94b77ca953cdc8b8fd08. I just have no idea how to test this for cloning.

Reply:
It's one of those things that realistically should never be hit during execution. However incase something goes wrong atleast we are attaching the errors which would display to the UI. It should be ok to not have testing for this